shared/src/sources/balancer/info_fetching.rs

```diff
@@ -5,17 +5,20 @@ use crate::{
     token_info::TokenInfoFetching, Web3,
 };
 use anyhow::{anyhow, Result};
-use contracts::{BalancerV2Vault, BalancerV2WeightedPool};
-use ethcontract::{batch::CallBatch, Bytes, H160, H256};
+use contracts::{BalancerV2StablePool, BalancerV2Vault, BalancerV2WeightedPool};
+use ethcontract::{batch::CallBatch, Bytes, H160, H256, U256};
 use mockall::*;
 use std::sync::Arc;
 
 #[derive(Clone)]
-pub struct WeightedPoolInfo {
+pub struct PoolInfo {
```

Comment:
I don't think this should be a single struct as the data is fundamentally different between the two kinds of pools. You can compose data if you want share some code:
```rust
pub struct CommonPoolInfo {
pub pool_id: H256,
    pub tokens: Vec<H160>,
    pub weights: Vec<Bfp>,
    pub scaling_exponents: Vec<u8>,
}

pub struct WeightedPoolInfo {
    common: CommonPoolInfo,
    weights: Vec<Bfp>,
}

pub struct StablePoolInfo {
    common: CommonPoolInfo,
    pub amplification_parameter: U256,
}
```